scipy/stats/stats.py

```diff
@@ -4541,6 +4542,131 @@ def ttest_rel(a, b, axis=0, nan_policy='propagate'):
+        that assumes equal population variances.
+        If False, perform Welch's t-test, which does not assume equal
+        population variance.
+
+    Returns
+    -------
+    statistic : float or array
+        t-statistic
+    pvalue : float or array
+        two-tailed p-value
+
+    
+    References
+    ----------
+    Karen K. Yuen (1974), The two-sample trimmed t for unequal population
+    variances, Biometrika Volume 61, Number 1, 165-170,
+    DOI: https://doi.org/10.1093/biomet/61.1.165
+
+    Examples
+    --------
+    
+    >>> from scipy import stats    
+
+    >>> a = (56, 128.6, 12, 123.8, 64.34, 78, 763.3)
+    >>> b = (1.1, 2.9, 4.2)
+
+    >>> t, p = stats.ttest_trimmed(a, b, equal_var=False)
+
+    (4.591598691181999, 0.00998909252078421)
+
+    """
+
+    a, b, axis = _chk2_asarray(a, b, axis)
+
+    # check both a and b
+    cna, npa = _contains_nan(a, nan_policy)
+    cnb, npb = _contains_nan(b, nan_policy)
+    contains_nan = cna or cnb
+    if npa == 'omit' or npb == 'omit':
+        nan_policy = 'omit'
+
+    if contains_nan and nan_policy == 'omit':
+        a = ma.masked_invalid(a)
+        b = ma.masked_invalid(b)
+        return mstats_basic.ttest_ind(a, b, axis, equal_var)
+
+    if a.size == 0 or b.size == 0:
+        return Ttest_indResult(np.nan, np.nan)
+
+    n1 = a.shape[axis]
+    n2 = b.shape[axis]
+
+    a.sort()
+    b.sort()
+
+    trimmed_index_a = math.floor((n1 / 100.0) * trimming_percentage)
+    trimmed_index_b = math.floor((n2 / 100.0) * trimming_percentage)
+
+    trimmed_n1 = n1 - (2 * trimmed_index_a)
+    trimmed_n2 = n2 - (2 * trimmed_index_b)
+
+    trimmed_mean_a = trim_mean(a, trimming_percentage / 100.0)
+    trimmed_mean_b = trim_mean(b, trimming_percentage / 100.0)
+
+    winsorized_variance_a = _calculate_winsorized_variance(trimmed_index_a, trimmed_n1, n1, a)
+    winsorized_variance_b = _calculate_winsorized_variance(trimmed_index_b, trimmed_n2, n2, b)
+
+    t = ((trimmed_mean_a - trimmed_mean_b) / (math.sqrt((winsorized_variance_a / trimmed_n1)
+                                                        + (winsorized_variance_b / trimmed_n2))))
+    df = math.pow(((winsorized_variance_a / trimmed_n1)
+                   + (winsorized_variance_b / trimmed_n2)), 2) / ((math.pow((winsorized_variance_a / trimmed_n1), 2)
+                                                                  / (trimmed_n1 - 1.0))
+                                                                  + (math.pow((winsorized_variance_b / trimmed_n2), 2)
+                                                                  / (trimmed_n2 - 1.0)))
+
+    t, prob = _ttest_finish(df, t)
+
+    return Ttest_relResult(t, prob)
+
+
+def _calculate_winsorized_variance(trimmed_index, trimmed_n, n, a):
+
+    trimmed_array = a[int(trimmed_index + 1):int(n - trimmed_index - 1)]
+    trimmed_sum = sum(trimmed_array)
+    winsorized_mean = (1.0 / n) * (((trimmed_index + 1.0) * a[int(trimmed_index)]) + trimmed_sum
```

Comment:
i think it would be better to construct a new vector a_trimmed and compute ist mean as np.meam(a_trimmed)